### ISSUE TITLE
flutter: Support web builds

### ISSUE THREAD

**Issue by GRBurst** [NixOS/nixpkgs#85574]
**Describe the bug**
`flutter-dev` crashes when trying to run a web project.

**To Reproduce**
Follow the few steps mentioned in the official documentation of flutter web:
https://flutter.dev/docs/get-started/web
Running `flutter run -d chrome` crashes with:
```
FileSystemException: FileSystemException: Failed to set file modification time, path = '/nix/store/8r1098995j256icfvidibwikcd2as10c-flutter-dev-1.17.0-dev.5.0-unwrapped/packages/flutter_tools/.packages' (OS Error: Read-only file system, errno = 30)
```

**Expected behavior**
`flutter-dev` serves app from localhost in Chromium.

**Additional context**
Here is the error log generated by `flutter-dev` when invoking `flutter-dev run -d chrome`
[flutter.log](https://github.com/NixOS/nixpkgs/files/4499956/flutter.log)

**Notify maintainers**

@babariviere 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
• No issues found!
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.28, NixOS, 20.09pre218613.ae6bdcc5358 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - channels(user): `"nixos-20.09pre220217.39247f8d04c, nixpkgs"`
 - channels(root): `"nixos-19.09-19.09.2370.e10c65cdb35, nixos-hardware"`
 - nixpkgs: `/home/user/.nix-defexpr/channels/nixpkgs

Maintainer information:
-

**Comment by babariviere** [comment]
Hello @GRBurst, thanks for your issue. After investigation, I found that the flutter is trying to rebuild its dependencies. One way to fix this would be to rebuild the package.

The nix `flutter` package is far from perfect. I would like to improve it but I don't know how to do it the proper way without taking a lot of time patching flutter source code.
This kind of behavior could be avoided by disabling versions dependencies and build a true "stateless" (I don't know how to say it haha) package.

I will try to disable dependencies check. It will take a little of my time. So, I won't be able to deliver a patch anytime soon, is it ok to you ?

**Comment by GRBurst** [comment]
> So, I won't be able to deliver a patch anytime soon, is it ok to you ?

@babariviere yeah sure - I am not in a hurry with this. Most important is that you are aware of this issue ;-)

> By the way, thank you for the log. It was really helpful !
:-)
